src/main/java/com/fasterxml/jackson/dataformat/xml/XmlFactory.java

```diff
@@ -665,6 +665,8 @@ protected FromXmlParser _createParser(byte[] data, int offset, int len, IOContex
             }
         } catch (XMLStreamException e) {
             return StaxUtil.throwAsParseException(e, null);
+        } catch (IndexOutOfBoundsException e) {
+            throw new JsonParseException(null, "Invalid input data.", e);
```

Comment:
Huh? No, this seems something that should not be needed: nothing ought to throw `IndexOutOfBoundsException`...